[Huang Hao] iP

huanghao1998 opened the pull request:
# Dudu
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Dudu frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is, 

1. download it from [here](https://github.com/huanghao1998/ip/releases/download/A-Jar/ip.jar). <sub>This is not updated version.<sub>
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😄 

And it is __FREE__!

Features:

- [x] Managing tasks
- [ ] Managing deadlines

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public static void main(String[] args) {
     new Dudu("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

huanghao1998 committed:
Add level-1 DODO

huanghao1998 committed:
Update UI text and divider

huanghao1998 committed:
Add, List

huanghao1998 committed:
Mark as Done

huanghao1998 committed:
ToDos, Events, Deadlines

huanghao1998 committed:
Automated Text UI Testing=

huanghao1998 committed:
Handle Errors

huanghao1998 committed:
Delete

huanghao1998 committed:
Use Enums and update Test case

huanghao1998 committed:
Revamp the code

huanghao1998 committed:
Regroup main java class

huanghao1998 committed:
Restruct code and complete save

huanghao1998 committed:
Add load task function

huanghao1998 committed:
Add date and time

huanghao1998 committed:
Merge level-7 and level-8

huanghao1998 committed:
Use more OOP

huanghao1998 committed:
Update .gitignore

huanghao1998 committed:
Merge branch add-gradle-support

huanghao1998 committed:
Add Junit test

huanghao1998 committed:
Add find function

huanghao1998 committed:
Delete data directory

Delete data directory

huanghao1998 committed:
Add A-Jar

huanghao1998 committed:
Add A-Jar

huanghao1998 committed:
Merge branch-Level-9 to master

huanghao1998 committed:
Update JavaDoc

huanghao1998 committed:
Add Parser class

huanghao1998 committed:
Improve Coding Standard

huanghao1998 committed:
Merge commit '384441619f5727b39f6f9e4ae2d0faaca306383a'

huanghao1998 committed:
Add more JTest and fix bug

huanghao1998 committed:
Delete META-INF file

huanghao1998 committed:
Add checkstyle, update the JavaDoc and the code style

huanghao1998 committed:
add checkstyle

huanghao1998 committed:
Use Varargs and update handling exception

huanghao1998 committed:
Add GUI for Dudu

huanghao1998 committed:
Merge branch-Level-10 to master

engenhui1999 commented on src/main/java/dudu/util/Parser.java:
```diff
@@ -0,0 +1,75 @@
+ * Parser class on parsing the input.
+ */
+public class Parser {
+    private enum Instruction {
```
I think your use of your instruction list is really neat!

engenhui1999 commented on src/main/java/dudu/util/Parser.java:
```diff
@@ -0,0 +1,75 @@
+            case HI:
+                return new IntroCommand();
+            case TODO:
+                if (input.trim().length() == 4) {
```
I like the way you use trim(), you manage to account for the whitespace before and after the command!

engenhui1999 commented on src/main/java/dudu/util/Storage.java:
```diff
@@ -0,0 +1,92 @@
+                list.add(Task.decode(scanner.nextLine()));
+            }
+        } catch (TaskIoException | FileNotFoundException | InvalidCommandException ex) {
+            System.out.println(ex);
```
Should this print line be further abstracted by the Ui class? I was thinking maybe you can have a Ui method that printError(DuduException error).

huanghao1998 replied:
I will implement it, thanks for the help 😁

engenhui1999 reviewed (commented):
Overall your code looks elegant! I like the way you used OOP to the fullest!

huanghao1998 committed:
Improve Gui and code style

EksdeeeX commented on src/main/java/dudu/command/Command.java:
```diff
@@ -0,0 +1,19 @@
+package dudu.command;
```
I like how you split all the commands into individual classes for better readability.

EksdeeeX commented on src/main/java/dudu/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+package dudu;
```
Great effort for implementing this GUI.

EksdeeeX commented on src/main/java/dudu/task/Task.java:
```diff
@@ -0,0 +1,135 @@
+     * @param task String of task information; separate using ",".
+     * @return Create a specific task using the String of task information.
+     */
+    public static Task decode(String task) throws InvalidCommandException {
```
neat and easily readable, good job

EksdeeeX commented on src/main/java/dudu/task/TaskList.java:
```diff
@@ -0,0 +1,107 @@
+ * Task list class for storing tasks
+ */
+public class TaskList {
+    private final ArrayList<Task> list;
```
perhaps a variable name stating the contents of the list would be helpful to others who are not familiar with your code? example "taskList".

EksdeeeX reviewed (commented):
overall a very clean and neat piece of code with cool unique features, good job!

huanghao1998 committed:
Use assert

Assertion: helps to check whether all the important objects are appeared when the method evoked

huanghao1998 committed:
Merge pull request #1 from huanghao1998/branch-A-Assertions

Use assert

huanghao1998 committed:
Check and improve code quality

Remove unwanted statements

huanghao1998 committed:
Merge pull request #2 from huanghao1998/branch-A-CodeQuality

Check and improve code quality

huanghao1998 committed:
Use Streams

huanghao1998 committed:
Merge pull request #3 from huanghao1998/branch-A-Streams

Use Streams

huanghao1998 committed:
Implement detect duplicates function

Add DuplicationException Class - when duplicates are detected, this will be thrown

huanghao1998 committed:
Merge pull request #4 from huanghao1998/branch-C-DetectDuplicates

Implement detect duplicates function

huanghao1998 committed:
Set up CI

huanghao1998 committed:
Improve Code standard

huanghao1998 committed:
Improve test code standard

huanghao1998 committed:
Add main method for Dudu

huanghao1998 committed:
Fix bug

huanghao1998 committed:
Improve comments

huanghao1998 committed:
Add backgound color for button and chatContainer

huanghao1998 committed:
Improve Gui so that it can be resizable with window

huanghao1998 committed:
Improve JavaDoc

huanghao1998 committed:
Merge pull request #6 from huanghao1998/branch-A-BetterGui

Branch a better gui

huanghao1998 committed:
Delete duplicate of file

huanghao1998 committed:
Change github CI file

huanghao1998 committed:
Uncomment the main function in Dudu

huanghao1998 committed:
Add credit

huanghao1998 committed:
Fix some bug on displaying to CLI, and partially done with Readme

huanghao1998 committed:
Update the readme

huanghao1998 committed:
Fix bug on Readme

huanghao1998 committed:
Fix checkstyle and format of README

huanghao1998 committed:
Complete the README

huanghao1998 committed:
Change mainClassName in build.gradle

huanghao1998 committed:
Improve README

huanghao1998 committed:
Improve README 2.0

huanghao1998 committed:
Improve README 3.0

huanghao1998 committed:
Update Main README

huanghao1998 committed:
Amend Main README arrangement

huanghao1998 committed:
Fix header comments
